FetchMigration/python/fetch_orchestrator.py

```diff
@@ -2,33 +2,100 @@
 __PIPELINE_OUTPUT_FILE_SUFFIX = "/pipelines/pipeline.yaml"
 
 
-def run(dp_base_path: str, dp_config_file: str, dp_endpoint: str) -> Optional[int]:
-    dp_exec_path = dp_base_path + __DP_EXECUTABLE_SUFFIX
-    output_file = dp_base_path + __PIPELINE_OUTPUT_FILE_SUFFIX
-    metadata_migration_params = MetadataMigrationParams(dp_config_file, output_file, report=True)
-    logging.info("Running pre-migration steps...\n")
+def __get_env_string(name: str) -> Optional[str]:
+    val: str = os.environ.get(name, "")
+    if len(val) > 0:
+        return val
+    else:
+        return None
+
+
+def update_target_host(dp_config: dict, target_host: str):
+    # Inline target host only supports HTTPS, so force it
+    target_with_protocol = target_host
+    match = __PROTOCOL_PREFIX_PATTERN.match(target_with_protocol)
+    if match:
+        target_with_protocol = target_host[match.end():]
+    target_with_protocol = __HTTPS_PREFIX + target_with_protocol
+    if len(dp_config) > 0:
+        # We expect the Data Prepper pipeline to only have a single top-level value
+        pipeline_config = next(iter(dp_config.values()))
+        # The entire pipeline will be validated later
+        if metadata_migration.SINK_KEY in pipeline_config:
+            # throws ValueError if no supported endpoints are found
+            plugin_name, plugin_config = metadata_migration.get_supported_endpoint(pipeline_config,
+                                                                                   metadata_migration.SINK_KEY)
+            plugin_config[metadata_migration.HOSTS_KEY] = [target_with_protocol]
+            pipeline_config[metadata_migration.SINK_KEY] = {plugin_name: plugin_config}
+
+
+def write_inline_pipeline(pipeline_file_path: str):
+    # This is expected to be a base64 encoded string
+    inline_pipeline = __get_env_string("INLINE_PIPELINE")
+    pipeline_yaml = yaml.safe_load(base64.b64decode(inline_pipeline))
+    inline_target_host = __get_env_string("INLINE_TARGET_HOST")
+    if inline_target_host is not None:
+        update_target_host(pipeline_yaml, inline_target_host)
+    with open(pipeline_file_path, 'w') as out_file:
+        # Note - this does not preserve comments
+        yaml.safe_dump(pipeline_yaml, out_file)
+
+
+def write_inline_target_host(pipeline_file_path: str, inline_target_host: str):
+    with open(pipeline_file_path, 'rw') as pipeline_file:
+        pipeline_yaml = yaml.safe_load(pipeline_file)
+        update_target_host(pipeline_yaml, inline_target_host)
+        # Note - this does not preserve comments
+        yaml.safe_dump(pipeline_yaml, pipeline_file)
+
+
+def run(params: FetchOrchestratorParams) -> Optional[int]:
+    # This is expected to be a base64 encoded string
+    inline_pipeline = __get_env_string("INLINE_PIPELINE")
+    inline_target_host = __get_env_string("INLINE_TARGET_HOST")
+    if inline_pipeline is not None:
+        write_inline_pipeline(params.pipeline_file_path)
```

Comment:
nit: seems like you could have passed the two env variables as parameters so this function didn't have to retrieve them

Reply:
Incorporated